eng/source-build-pre-release.yml

```diff
@@ -0,0 +1,146 @@
+  - job: GetBuildInfoJob
+    displayName: Get Build Info
+    steps:
+    - download: dotnet-staging-pipeline-resource
+      artifact: manifests
+      patterns: 'manifest.json'
+      displayName: Download Release Manifest
+    - download: dotnet-staging-pipeline-resource
+      artifact: drop
+      patterns: 'config.json'
+      displayName: Download Release Config
+    - script: |
+        set -euo pipefail
+
+        manifest_path=$(PIPELINE.WORKSPACE)/dotnet-staging-pipeline-resource/manifests/manifest.json
+        config_path=$(PIPELINE.WORKSPACE)/dotnet-staging-pipeline-resource/drop/config.json
+
+        runtime_version="$(jq -r '.Runtime' $config_path)"
+        release_channel="$(jq -r '.Channel' $config_path)"
+
+        # Source-build only supports the lowest avaiable feature band.
+        # Sort the SDK releases by number and pick the lowest value.
+        sdk_version="$(jq -r '.Sdks | sort_by(. | split(".") | map(tonumber)) | .[0]' $config_path)"
+
+        # replace the last two characters in sdk_version with xx
+        branch_version=$(echo $sdk_version | sed 's/..$/xx/')
+        branch_name="refs/heads/internal/release/${branch_version}"
+
+        commit=$(jq -r --arg BRANCH "${branch_name}" '.builds[] | select(.repo == "https://dev.azure.com/dnceng/internal/_git/dotnet-installer") | select(.branch == $BRANCH) | .commit' $manifest_path)
+
+        if [[ ! $commit ]]; then
+          echo "Installer commit not set. Exiting..."
+          exit 1
+        fi
+
+        echo "Release channel: ${release_channel}"
+        echo "Runtime version: ${runtime_version}"
+        echo "SDK version: ${sdk_version}"
+        echo "Installer branch: ${branch_name}"
+        echo "Installer commit: ${commit}"
+
+        echo "##vso[task.setvariable variable=SdkVersion;isOutput=true]${sdk_version}"
+        echo "##vso[task.setvariable variable=RuntimeVersion;isOutput=true]${runtime_version}"
+        echo "##vso[task.setvariable variable=ReleaseChannel;isOutput=true]${release_channel}"
+        echo "##vso[task.setvariable variable=InstallerCommit;isOutput=true]${commit}"
+      name: ReadReleaseInfoStep
+      displayName: Read Release Info
+
+- stage: UploadSourceTarballStage
+  displayName: Upload Source Tarball
+  dependsOn: GetBuildInfoStage
+  jobs:
+  - job: UploadSourceTarballJob
+    displayName: Upload Source Tarball
+    variables:
+    - name: InstallerCommit
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.InstallerCommit'] ]
+    - name: SdkVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.SdkVersion'] ]
+    - name: RuntimeVersion
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.RuntimeVersion'] ]
+    - name: ReleaseChannel
+      value: $[ stageDependencies.GetBuildInfoStage.GetBuildInfoJob.outputs['ReadReleaseInfoStep.ReleaseChannel'] ]
+    steps:
+    - script: |
```

Comment:
Is this step necessary?  If seems like a step used to developer the initial pipeline.